src/python/gudhi/barycenter.py

```diff
@@ -0,0 +1,233 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Theo Lacombe
+#
+# Copyright (C) 2019 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+
+import ot
+import numpy as np
+import scipy.spatial.distance as sc
+
+from gudhi.wasserstein import _build_dist_matrix, _perstot
+
+
+def _proj_on_diag(w):
```

Comment:
You already have that one in gudhi.wasserstein.

Reply:
Actually they take slightly different objects, the one here returns the projection onto the diagonal of a single point, while the one in `gudhi.wasserstein` does the same for an array of points. 

However, this one is only used once so I just removed it and slightly changed the code in `_mean`.